plugins/contentful/src/plugin.ts

```diff
@@ -0,0 +1,119 @@
+import { registerDataPlugin } from '@builder.io/data-plugin-tools';
+import pkg from '../package.json';
+import contentful from 'contentful';
+import qs from 'qs';
+// Define Commerce.js plugin ID. ID string should match package name.
+const pluginId = pkg.name;
+
+registerDataPlugin(
+  {
+    id: pluginId,
+    name: 'Contentful',
+    icon:
+      'https://cdn.builder.io/api/v1/image/assets%2FYJIGb4i01jvw0SRdL5Bt%2Fd6097cd40fef4b94b18a3e0c4c53584d',
+    settings: [
+      {
+        name: 'spaceId',
+        type: 'string',
+        required: true,
+        helperText: 'TODO',
+      },
+      {
+        name: 'accessToken',
+        type: 'string',
+        required: true,
+        helperText: 'TODO',
+      },
+    ],
+    ctaText: `Connect your Contentful space`,
+  },
+  // Observable map of the settings configured above
+  async settings => {
+    // Get public key input from user
+    const spaceId = settings.get('spaceId')?.trim();
+    const accessToken = settings.get('accessToken')?.trim();
+    const client = await contentful.createClient({
+      space: spaceId,
+      accessToken,
+    });
+    console.log(' here contentful ', client);
+    return {
+      async getResourceTypes() {
+        const contentTypes = await client.getContentTypes();
+        console.log(' here content types are ', contentTypes);
+        return contentTypes.items.map(type => ({
+          name: type.name,
+          id: type.sys.id,
+          canPickEntries: true,
+          inputs: () => {
+            const fields = [
+              {
+                name: 'include',
+                friendlyName: 'limit',
+                defaultValue: 10,
+                type: 'number',
+              },
+            ];
+            const acceptableFields = type.fields.filter(field =>
+              ['Text', 'Boolean', 'Number', 'Symbol'].includes(field.type)
+            );
+            if (acceptableFields.length > 0) {
+              fields.push({
+                name: 'fields',
+                type: 'object',
+                friendlyName: `${type.name} fields`,
+                subFields: acceptableFields.map(field => ({
+                  ...field,
+                  type: field.type === 'Symbol' ? 'Text' : field.type.toLowerCase(),
+                  name: field.id,
+                  friendlyName: field.name,
+                })),
+              } as any);
+            }
```

Comment:
ooo what does this `fields` stuff do?

Reply:
it allows querying model by fields, to form a query at the end `&fields.firstName=Steve`, recording a demo to showcase this

Reply:
nice!!!!!! I thought that would end up being really hard to support, but this has huge value

how hard would it be to add this to contentstack too @samijaber ?

and is there an easy way to pick a field to sort on? if we can have basic querying by fields and basic sorting by fields that covers a ton of use cases and where I was hoping we'd get to (but didn't think we'd be able to so soon!!)

Reply:
Very cool! Will connect w/ Aziz about this (please share your demo whenever), as I avoided digging into how these `inputs` and decided to leave it for later.

Will merge my v1 of the plugin as-is, and look into improving querying/filtering/searching in a 2nd pass.

Reply:
shared demo in slack

Reply:
<img width="454" alt="Screen Shot 2021-09-20 at 2 31 34 PM" src="https://user-images.githubusercontent.com/5093430/134078687-694600f2-2d4c-48c4-94f3-9482e8a24f6e.png">
a
<img width="451" alt="Screen Shot 2021-09-20 at 2 31 40 PM" src="https://user-images.githubusercontent.com/5093430/134078694-1a2f1ecf-cbe9-491b-8818-e19e622550b0.png">
added a field to order results, and hid the fields query as advanced with some helper texts